SpatialGDK/Source/SpatialGDK/Public/Interop/RPCs/RPCTypes.h

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "Interop/Connection/SpatialGDKSpanId.h"
+#include "Schema/RPCPayload.h"
+#include "SpatialView/EntityView.h"
+
+namespace SpatialGDK
+{
+struct RPCReadingContext
+{
+	Worker_EntityId EntityId;
+	Worker_ComponentId ComponentId;
+
+	bool IsUpdate() const { return Update != nullptr; }
+
+	Schema_ComponentUpdate* Update = nullptr;
+	Schema_Object* Fields = nullptr;
+};
+
+struct RPCWritingContext
+{
+	enum class DataKind
+	{
+		Generic,
+		ComponentData,
+		ComponentUpdate,
+		CommandRequest,
+		CommandResponse
+	};
+
+	class EntityWrite
```

Comment:
This seems like a fairly unusual class structure. It's a nested class that's moveable, but not move assignable and it does a lot of it's work in the destructor. It might be worth a comment explaining why it works like this.

Reply:
Good point in general, I just completely forgot comments in this whole PR.